src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/UserMappingServiceTest.kt

```diff
@@ -0,0 +1,167 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.integration
+
+import io.mockk.every
+import io.mockk.mockk
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.CsvSource
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.factory.ApAreaEntityFactory
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.factory.ProbationRegionEntityFactory
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.factory.StaffUserDetailsFactory
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.factory.StaffUserTeamMembershipFactory
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.ApAreaRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.UserMappingService
+
+class UserMappingServiceTest {
+
+  private val apAreaRepository = mockk<ApAreaRepository>()
+
+  private val service = UserMappingService(apAreaRepository)
+
+  private val usersProbationRegionArea = ApAreaEntityFactory().produce()
```

Comment:
would something like `usersApAreaThroughProbationRegion` or `usersProbationRegionApArea` be clearer here? I think using "Area" on it's own is inherently confusing as we have Delius "Probation area" and CAS1 "AP Area"...